internal/backup_detail.go

```diff
@@ -0,0 +1,25 @@
+package internal
+
+// BackupDetails is used to append ExtendedMetadataDto details to BackupTime struct
+type BackupDetail struct {
+	BackupTime
+	ExtendedMetadataDto
+}
+
+// writeBackupTime sets matching elements in BackupDetail with values from BackupTime
+func (backupDetail *BackupDetail) writeBackupTime(backupTime BackupTime) {
```

Comment:
Let's remove these two functions. Instead you can construct such an object with 'BackupDetail{backupTimeObj, ExtendedMetadataDtoObj}' statement.

Reply:
Thanks for feedback.. still learning GO.. fixed it as requested ..

Reply:
Nice, thanks!